assets/app/view/game/round/stock.rb

```diff
@@ -296,15 +296,19 @@ def render_bank_companies
 
           @game.companies.select { |c| c.owner == @game.bank }.map do |company|
             children = []
-            children << h(Company, company: company)
-            children << h('div.margined_bottom', { style: { width: '20rem' } },
-                          render_buy_input(company)) if @selected_company == company
+            children << h(Company, company: company,
+                                   bids: (@current_actions.include?('bid') ? @step.bids[company] : nil))
+            if @selected_company == company
+              inputs = []
+              inputs.concat(render_buy_input(company)) if @current_actions.include?('buy_company')
+              inputs.concat(render_bid_input(company)) if @current_actions.include?('bid')
+              children << h('div.margined_bottom', { style: { width: '20rem' } }, inputs)
+            end
             h(:div, props, children)
           end
         end
 
         def render_buy_input(company)
-          return [] unless @current_actions.include?('buy_company')
```

Comment:
Why did you remove this line?

Reply:
I moved it to row 303 to be more cohesive with the bid action i added.

```ruby
inputs.concat(render_buy_input(company)) if @current_actions.include?('buy_company')
inputs.concat(render_bid_input(company)) if @current_actions.include?('bid')
```